Expecto.Tests/Tests.fs

```diff
@@ -771,24 +771,22 @@ let stress =
       ]
 
     let deadlockTest() =
-      let lockOne = obj()
-      let lockTwo = obj()
+      let lockOne = new System.Threading.ReaderWriterLockSlim()
+      let lockTwo = new System.Threading.ReaderWriterLockSlim()
       testList "deadlock" [
         testCaseAsync "case A" <| async {
-          lock lockOne (fun () ->
-            Thread.Sleep 10
-            lock lockTwo (fun () ->
-              ()
-            )
-          )
+          lockOne.EnterReadLock()
+          Thread.Sleep 100
```

Comment:
May I suggest that you fire off a semaphore in A, that you WaitOne on in B? That way you can easily sequence it so that both A and B have taken the read lock before they try to acquire the write lock.